ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx

```diff
@@ -370,7 +371,9 @@ function WorkloadCvesOverviewPage() {
     );
 
     return (
-        <>
+        <AutocompleteContextProvider
+            value={isViewingWithCves ? { 'Image CVE Count': ['>0'] } : { 'Image CVE Count': ['0'] }}
+        >
```

Comment:
I don't really think Context is the best fit for this functionality, for a handful of reasons:

1. It introduces state to something that we can fairly easily derive from the current values in scope. Instead of the context with a getter/setter, we could just pass `{isViewingWithCves ? { 'Image CVE Count': ['>0'] } : { 'Image CVE Count': ['0'] }}` directly to the filter component itself instead of using this value as the default state. We would then automatically get the updates when `isViewingWithCves` changes instead of needing to manually set the next value in the `ObservedCveModeSelect` component. As we add more to the autocomplete context, i.e. the current searchFilter, other page specific contexts..., it will be easy for situations to arise where these fall out of sync.
2. IMO the prop drilling is fairly limited in this case. We will need to pass the context to the top level `CompoundSearchFilter`, which will need to pass it through its child tree, but this is something we should only need to worry about once. This isn't a value that will need to be read at varied depths through our component hierarchy. Once the initial implementation is complete we only need to worry about passing the value to the top level component as a prop, which will be a similar amount of overhead as creating a ContextProvider each time.
3. I think the Context with state adds a little too much indirection, and co-mingles concerns in components that do not need to know about the autocomplete functionality. In this case, the `ObservedCveModeSelect` really only needs to care about setting the CveMode, and deriving the value like mentioned in # 1 above means we can keep these components focused on their single responsibility.

Reply:
1+2: Hmm you have a fair point. There are a few levels of prop drilling, but it might not be as significant as I initially thought. If we opt for prop drilling, we wouldn't need to maintain state, which would be beneficial. As for incorporating the current search filter, I'm still sure about the idea, but I'll check with UX to get their perspective. If this concern isn't as problematic as it seems, we can see how to make it more extensible.

3: Yea, that's also a good point. There was a bit of co-mingling of concerns because of the Context implementation. Let's try the prop method and see how it goes